tensorflow/c/kernels_test.cc

```diff
@@ -161,6 +161,328 @@ TEST(TestKernel, TestRegisterKernelBuilder) {
   ASSERT_TRUE(delete_called);
 }
 
+// REGISTER_OP for TF_OpKernelConstruction_GetAttr* test cases.
+// Registers two ops, each with a single attribute called 'Attr'.
+// The attribute in one op will have a type 'type', the other
+// will have list(type).
+#define ATTR_TEST_REGISTER_OP(name, type)                     \
+  REGISTER_OP("TestKernelAttr" #name)                         \
+      .Attr("Attr: " #type)                                   \
+      .SetShapeFn(tensorflow::shape_inference::UnknownShape); \
+  REGISTER_OP("TestKernelAttr" #name "List")                  \
+      .Attr("Attr: list(" #type ")")                          \
+      .SetShapeFn(tensorflow::shape_inference::UnknownShape)
+ATTR_TEST_REGISTER_OP(String, string);
+ATTR_TEST_REGISTER_OP(Int, int);
+ATTR_TEST_REGISTER_OP(Float, float);
+ATTR_TEST_REGISTER_OP(Bool, bool);
+ATTR_TEST_REGISTER_OP(Type, type);
+#undef ATTR_TEST_REGISTER_OP
+
+// Helper macros for the TF_OpKernelConstruction_GetAttr* tests.
+#define EXPECT_TF_META(attr_name, expected_list_size, expected_type,          \
+                       expected_total_size)                                   \
+  do {                                                                        \
+    auto m = TF_OpKernelConstruction_GetAttrMetadata(ctx, attr_name, status); \
+    EXPECT_EQ(TF_OK, TF_GetCode(status)) << TF_Message(status);               \
+    const unsigned char e = expected_list_size >= 0 ? 1 : 0;                  \
```

Comment:
Can you change `e` to a more descriptive name? Perhaps `should_be_list`

Reply:
done